metis-media-service/src/main/java/eu/europeana/metis/mediaprocessing/extraction/ThumbnailGenerator.java

```diff
@@ -106,22 +118,25 @@ private static synchronized Path initColorMap() throws MediaProcessorException {
     return globalColormapFile;
```

Comment:
Yes, this is annoying. And again this is the way it was done before, but there are reasons. 

This file is used by ImageMagick. It is passed in the form of a file path in the big ImageMagick command that is created in this class. So we cannot use the original: this could be embedded in a jar file. Hence we make a temporary copy of the file that we can use in the image magick command and which will be destroyed when the VM exits. I have not been able to find out definitively whether we can also send a stream to image magick instead of the file, which would be the alternative (and I didn't want to spend too much time on that).

The duplicate color map file exists because we only want to copy the files once (for all instances of the thumbnail generator). This would be enough, and an instance could refer to this file, except that we need to give the class all resources it needs in a non-static way so that we can properly mock and test it.

I hope this explains?